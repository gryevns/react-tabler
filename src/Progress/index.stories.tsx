import React from "react";

import Progress from "../Progress";

export default {
    title: "Progress",
};

export const Default = () => (
    <Progress size="md">
        <Progress.Bar color="blue" width={66} />
    </Progress>
);

export const Indeterminate = () => (
    <Progress>
        <Progress.Bar indeterminate />
    </Progress>
);

export const Colors = () => (
    <>
        <Progress size="sm" className="mb-2">
            <Progress.Bar color="red" width={24} />
        </Progress>
        <Progress size="sm" className="mb-2">
            <Progress.Bar color="green" width={45} />
        </Progress>
        <Progress size="sm" className="mb-2">
            <Progress.Bar color="purple" width={64} />
        </Progress>
        <Progress size="sm" className="mb-2">
            <Progress.Bar indeterminate color="pink" />
        </Progress>
    </>
);
