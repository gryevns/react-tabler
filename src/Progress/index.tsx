import * as React from "react";
import cn from "classnames";

import { Color } from "../common";

export interface ProgressBarProps {
    className?: string;
    color?: Color;
    width?: number;
    indeterminate?: boolean;
    style?: React.CSSProperties;
}

const ProgressBar = ({
    className,
    color,
    indeterminate = false,
    width = 0,
    style,
    ...rest
}: ProgressBarProps) => {
    const classes = cn(
        "progress-bar",
        {
            [`bg-${color}`]: !!color,
            "progress-bar-indeterminate": indeterminate,
        },
        className,
    );
    return (
        <div
            className={classes}
            style={{ width: `${width}%`, ...style }}
            {...rest}
            role="progressbar"
            aria-valuenow={width}
            aria-valuemin={0}
            aria-valuemax={100}
        >
            <span className="sr-only">{width}% Complete</span>
        </div>
    );
};

export interface ProgressProps {
    className?: string;
    children?: React.ReactNode;
    card?: boolean;
    size?: "xs" | "sm" | "md";
}

function Progress({ className, children, card, size, ...rest }: ProgressProps) {
    const classes = cn(
        `progress`,
        { [`progress-${size}`]: !!size, "card-progress": card },
        className,
    );
    return (
        <div className={classes} {...rest}>
            {children}
        </div>
    );
}

Progress.Bar = ProgressBar;

export default Progress;
