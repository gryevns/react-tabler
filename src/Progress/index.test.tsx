import React from "react";
import { shallow } from "enzyme";

import Progress from "./index";

describe("Progress", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
            <Progress size="md">
                <Progress.Bar width={66} />
            </Progress>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match in snapshot", () => {
        const wrapper = shallow(<Progress.Bar color="red" indeterminate />);
        expect(wrapper).toMatchSnapshot();
    });
});
