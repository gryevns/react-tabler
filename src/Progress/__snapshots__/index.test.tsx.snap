// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match in snapshot 1`] = `
<div
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="progress-bar bg-red progress-bar-indeterminate"
  role="progressbar"
  style={
    Object {
      "width": "0%",
    }
  }
>
  <span
    className="sr-only"
  >
    0
    % Complete
  </span>
</div>
`;

exports[`Progress should match snapshot 1`] = `
<div
  className="progress progress-md"
>
  <ProgressBar
    width={66}
  />
</div>
`;
